### PR TITLE
Maybeboard feature.

### LINKED ISSUE #246: A maybeboard to hold potential cube inclusions
Have a dedicated area to put potential inclusions for the cube so that I don't forget about it. This area would not count towards any of the cube's statistics, but would be displayed on the cube listing page. Controls to easily move cards to and from this area would also be provided.

Currently users can use blog posts and the overview to track maybeboard cards. So this is not a pressing feature.

### PR DESCRIPTION
**Pull request by phulin**
This PR adds a "maybeboard" where you can store cards you are considering adding to a cube. The maybeboard is hidden by default and is exposed by an option in the "Display" menu. Clicking on a card in the maybeboard adds it to a changelist so you can add it to the cube.

Screenshot:
image

Closes #246.

### CODE REVIEW

File: routes/cube_routes.js [hunk 1]
```diff
@@ -3160,6 +3165,65 @@ router.post('/api/updatecards/:id', ensureAuth, function(req, res) {
   });
 });
 
+function maybeCards(cube) {
+  const maybe = (cube.maybe || []).filter((card) => card.cardID);
```
Comment (anderoonies) [inline routes/cube_routes.js]: are there negative or 0 cardIDs you're filtering out? do those just exist in the cube?

File: models/cube.js [hunk 2]
```diff
@@ -50,20 +63,8 @@ const cubeSchema = mongoose.Schema({
...
-      cmc: Number,
-      cardID: String,
-      type_line: String,
-      addedTmsp: Date,
-      imgUrl: String,
-      details: {},
-    },
-  ],
+  cards: [Card],
+  maybe: [Card],
```
Comment (dekkerglen) [inline models/cube.js]: I'd like to include default values for these

File: routes/cube_routes.js [hunk 3]
```diff
@@ -3160,6 +3165,65 @@ router.post('/api/updatecards/:id', ensureAuth, function(req, res) {
...
+        success: 'false',
+        message: 'Cube not found.',
+      });
+    }
+
+    return res.status(200).send({
+      success: 'true',
+      maybe: maybeCards(cube),
+    });
+  }),
```
Comment (dekkerglen) [inline routes/cube_routes.js]: can we wrap this in a try/catch?

File: routes/cube_routes.js [hunk 4]
```diff
@@ -3160,6 +3165,65 @@ router.post('/api/updatecards/:id', ensureAuth, function(req, res) {
...
+    const withAdded = [...withRemoved, ...addCardsNoDetails];
+
+    cube.maybe = withAdded;
+    await cube.save();
+
+    return res.status(200).send({
+      success: 'true',
+    });
+  }),
+);
```
Comment (dekkerglen) [inline routes/cube_routes.js]: I would also like to wrap this in a try/catch

File: routes/cube_routes.js [hunk 5]
```diff
@@ -3160,6 +3165,65 @@ router.post('/api/updatecards/:id', ensureAuth, function(req, res) {
...
+  '/api/maybe/:id',
+  ensureAuth,
+  util.wrapAsyncApi(async function(req, res) {
+    const cube = await Cube.findOne(build_id_query(req.params.id));
+    if (!cube) {
+      return res.status(404).send({
+        success: 'false',
+        message: 'Cube not found.',
+      });
+    } else if (!req.user._id.equals(cube.owner)) {
```
Comment (dekkerglen) [inline routes/cube_routes.js]: else not needed, as we return in the previous if block

File: routes/cube_routes.js (hunk 3 above)
Comment (dekkerglen) [reply routes/cube_routes.js]: nvm - I just saw the wrapper method

File: routes/cube_routes.js (hunk 4 above)
Comment (dekkerglen) [reply routes/cube_routes.js]: nvm - I just saw the wrapper method

File: src/components/EditCollapse.js [hunk 6]
```diff
@@ -11,6 +11,24 @@ import ChangelistContext from './ChangelistContext';
...
+  if (name && name.length > 0) {
+    const normalized = encodeName(name);
+    const response = await fetch(`/cube/api/getcard/${normalized}`);
+    if (!response.ok) {
+      setAlerts((alerts) => [...alerts, { color: 'danger', message: `Couldn't get card: ${response.status}.` }]);
+      return null;
+    }
+
+    const json = await response.json();
+    if (json.success !== 'true' || !json.card) {
```
Comment (dekkerglen) [inline src/components/EditCollapse.js]: In what situation will only one of these conditionals be true? Would it make sense to just use one if statement here?

File: routes/cube_routes.js (hunk 1 above)
Comment (phulin) [reply routes/cube_routes.js]: No, just trying to avoid server crashes if something gets corrupted.

File: src/components/EditCollapse.js (hunk 6 above)
Comment (phulin) [reply src/components/EditCollapse.js]: I don't think there is one, just trying to be robust to weird failures + bugs

File: routes/cube_routes.js (hunk 5 above)
Comment (phulin) [reply routes/cube_routes.js]: just habit and it lets me put it on the same line

File: models/cube.js (hunk 2 above)
Comment (phulin) [reply models/cube.js]: done

**Pull request merged**